assets/js/components/page-selector/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { PanelBody, SelectControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useEditorContext } from '@woocommerce/base-context';
+/**
+ * Internal dependencies
+ */
+import { formatTitle } from '../utils';
+
+const PageSelector = ( { setPageId, pageId, defaultPageId, labels } ) => {
+	const { currentPostId } = useEditorContext();
+	const pages =
+		useSelect( ( select ) => {
+			return select( 'core' ).getEntityRecords( 'postType', 'page', {
+				status: 'publish',
+				orderby: 'title',
+				order: 'asc',
+				per_page: 100,
+			} );
+		}, [] ) || null;
+	// We should not render this if this page is the default one page
+	// and the link to option is set to default.
+	if ( ! ( currentPostId === defaultPageId && pageId === 0 ) && pages ) {
```

Comment:
I think having the above logic _here_ limits reuse of the page selector. This use case is going to be specific to cart and checkout blocks. Can we move it to the block's edit.js instead? Including the use of editor context.